cmd/prometheus/main.go

```diff
@@ -94,13 +97,8 @@ func main() {
 
 	a.HelpFlag.Short('h')
 
-	a.Flag("log.level",
-		"Only log messages with the given severity or above. One of: [debug, info, warn, error, fatal]").
-		Default("info").StringVar(&cfg.logLevel)
-
-	a.Flag("log.format",
-		`Set the log target and format. Example: "logger:syslog?appname=bob&local=7" or "logger:stdout?json=true"`).
-		Default("logger:stderr").StringVar(&cfg.logFormat)
+	a.Flag(promlog.LevelFlagName, promlog.LevelFlagHelp).
```

Comment:
I've been thinking a bit about a middle ground.  
The issue of easily introducing breaking changes and having to introduce weird forks is fixed. For many cases there is simply no significantly better way than just duplicating some code, and that's fine IMO.

For the problem of plugging together the core plumbing here across components: I think the right solution is to write a skeleton generator that creates the basic Go files. That would be generally quite useful for people to quickly write their own exporters.  
We can than add new flags etc. to the generated files. Then the generator can just be re-run against a repo and one can easily verify potential breaking changes by running `git diff` and remove them if necessary.  
There should be generally no need to modify the generated structures significantly so no hard-to-resolve diffs should occur.

That seems the right tooling to solve the problem here and generally allows us to easily apply such changes to a wide range of repos.